sycl/plugins/cuda/pi_cuda.cpp

```diff
@@ -5589,6 +5589,58 @@ pi_result cuda_piGetDeviceAndHostTimer(pi_device Device, uint64_t *DeviceTime,
   return PI_SUCCESS;
 }
 
+pi_result cuda_piextEnablePeer(pi_device command_device,
+                               pi_device peer_device) {
+
+  pi_result result = PI_SUCCESS;
+  try {
+    ScopedContext active(command_device->get_context());
+    result =
+        PI_CHECK_ERROR(cuCtxEnablePeerAccess(peer_device->get_context(), 0));
+
+  } catch (pi_result err) {
+    result = err;
+  }
+  return result;
+}
+
+pi_result cuda_piextDisablePeer(pi_device command_device,
+                                pi_device peer_device) {
+
+  pi_result result = PI_SUCCESS;
+  try {
+    ScopedContext active(command_device->get_context());
+    result = PI_CHECK_ERROR(cuCtxDisablePeerAccess(peer_device->get_context()));
+
+  } catch (pi_result err) {
+    result = err;
+  }
+  return result;
+}
+
+pi_result cuda_piextCanAccessPeer(pi_device command_device,
+                                  pi_device peer_device, pi_peer_attr attr) {
+
+  int value;
+  pi_result result = PI_SUCCESS;
+
+  CUdevice_P2PAttribute CUattr =
+      attr == access_supported
+          ? CU_DEVICE_P2P_ATTRIBUTE_ACCESS_SUPPORTED
+          : CU_DEVICE_P2P_ATTRIBUTE_NATIVE_ATOMIC_SUPPORTED;
+  try {
+    ScopedContext active(command_device->get_context());
+    PI_CHECK_ERROR(cuDeviceGetP2PAttribute(
+        &value, CUattr, command_device->get(), peer_device->get()));
+  } catch (pi_result err) {
+    result = err;
+  }
+  if (value == 0) {
```

Comment:
According to the [CUDA docs](https://docs.nvidia.com/cuda/cuda-driver-api/group__CUDA__PEER__ACCESS.html#group__CUDA__PEER__ACCESS_1g4c55c60508f8eba4546b51f2ee545393), it seems like only the value 1 is specified to have a specific meaning for these attributes, so I think that should be what we check instead:
```suggestion
  if (value != 1) {
```

Reply:
good catch thanks!